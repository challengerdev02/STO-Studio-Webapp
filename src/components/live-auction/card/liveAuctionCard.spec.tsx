import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { LiveAuctionCard } from './index';

describe('Live Auction Card component', function () {
  afterEach(cleanup);
  let props = {
    itemCoverImage: '/zachary-kadolph-unsplash.svg',
    itemName: 'Crypto Bull Society',
    bestOffer: '0.51',
    lastOffer: '0.42',
    likes: '1.7 k',
    onClick: () => {},
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<LiveAuctionCard {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('should test the LiveAuctionCard component displays ', () => {
    const { getByTestId } = setup();
    expect(getByTestId('live-auction-card')).toBeInTheDocument();
  });

  it('should test the source attribute of LiveAuctionCard component cover image has a value ', () => {
    const { getAllByRole } = setup();
    const cardImage = getAllByRole('img');
    expect(cardImage[0]).toHaveAttribute(
      'src',
      '/zachary-kadolph-unsplash.svg'
    );
    expect(cardImage[0]).toHaveAttribute('alt', '');
  });

  it('should test the live auction card received an item name props', () => {
    const { getByTestId } = setup();
    const itemName = getByTestId('item-name');
    expect(itemName).toBeInTheDocument();
    expect(itemName).toHaveTextContent(/Crypto Bull Society/i);
  });

  it('should test the live auction card received a best offer props', () => {
    const { getByTestId } = setup();
    const bestoffer = getByTestId('best-offer');
    expect(bestoffer).toBeInTheDocument();
    expect(bestoffer).toHaveTextContent(/0.51/i);
  });

  it('should test the live auction card received a last offer props', () => {
    const { getByTestId } = setup();
    const lastOffer = getByTestId('last-offer');
    expect(lastOffer).toBeInTheDocument();
    expect(lastOffer).toHaveTextContent(/0.42/i);
  });

  it('should test the live auction card received a likes props', () => {
    const { getByTestId } = setup();
    const likes = getByTestId('likes');
    expect(likes).toBeInTheDocument();
    expect(likes).toHaveTextContent(/1.7 k/i);
  });

  it('should test the live auction card has an options widget', () => {
    const { getByTestId } = setup();
    const optionsWidget = getByTestId('options-widget');
    expect(optionsWidget).toBeInTheDocument();
  });
});
