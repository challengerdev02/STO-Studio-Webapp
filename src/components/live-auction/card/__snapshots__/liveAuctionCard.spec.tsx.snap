// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Live Auction Card component should match with snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL LPZmN"
    data-testid="live-auction-card"
  >
    <div
      class="sc-gsDKAQ iCkGBE"
    >
      <img
        alt=""
        class="sc-dkPtRN cxcUvX"
        src="/zachary-kadolph-unsplash.svg"
      />
    </div>
    <div
      class="sc-hKwDye iPxAyv"
    >
      <div
        class="card-content-inner"
      >
        <div
          class="left-col"
        >
          <span
            class="left-col-row"
            title="Crypto Bull Society"
          >
             
            Crypto Bull Society
          </span>
          <span
            class="left-col-row left-col-row-variant"
            data-testid="item-name"
            title="Crypto Bull Society"
          >
            Crypto Bull Society
          </span>
        </div>
        <div
          class="right-col"
        >
          <div
            class="row-wrapper"
          >
            <span
              class="right-col-row"
            >
              <div
                class="sc-jRQBWg cSbcyc"
              >
                Best Offer
              </div>
            </span>
            <span
              class="right-col-row "
            >
              <div
                class="sc-gKclnd hcEPWM"
                data-testid="best-offer"
              >
                <span
                  class="last-typeof-right-col-row"
                >
                  <img
                    alt=""
                    class="sc-iCfMLu"
                    src="/test-file-stub"
                  />
                  0.51
                </span>
              </div>
            </span>
          </div>
          <div
            class="row-wrapper"
          >
            <div
              class="sc-gKclnd hcEPWM"
              data-testid="last-offer"
            >
              <div
                class="sc-jRQBWg cSbcyc"
              >
                Last
              </div>
               
              <span
                class="last-typeof-right-col-row"
              >
                 
                <img
                  alt=""
                  class="sc-iCfMLu"
                  src="/test-file-stub"
                />
                0.42
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-divider ant-divider-horizontal sc-furwcr jHNOkh"
        role="separator"
      />
      <div
        class="sc-eCImPb Snept"
      >
        <div
          class="sc-gKclnd hcEPWM"
        >
          <span
            class="sc-pVTFL FLoRu"
            data-testid="options-widget"
          >
            ...
          </span>
        </div>
         
        <div
          class="sc-gKclnd hcEPWM"
          data-testid="likes"
        >
           
          <span
            class="last-typeof-right-col-row likes"
          >
            <img
              class="sc-iCfMLu"
              src="/test-file-stub"
            />
            1.7 k
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
